weather_app_functions.R

```diff
@@ -0,0 +1,234 @@
+### weather_app in R Shiny ###
+
+library(httr)
+library(jsonlite)
+library(dplyr)
+library(leaflet)
+library(mapview)
+library(magick)
+library(tidygeocoder)
+
+# tidygeocoder::geo_osm() uses a string with an address as input and returns
+# the latitude and longitude of the location. 
+
+# you will need an API key to access the data from openweathermaps.org 
+# you have to sign up for a free account to get the api key 
+# the api key is private and should not be shared with other people
+
+my_api_key <- "insert your api key here" 
+
+
+# The following function takes the location (in string format) and your apiKey (in string format) as input.
+# The function returns the weather data for a specific location as a list.
+# The weather data are provided by www.openweathermap.org
+
+get_weather <- function(location, apiKey) {
+  
+  # first, we need to use tidygeocoder::geo_osm to get the longitude and the latitude of the desired location. 
+  
+  latitude <- tidygeocoder::geo_osm(location)$lat[1]
+  longitude <- tidygeocoder::geo_osm(location)$long[1]
+  
+  # now, we access www.openweatherapp.org and retrieve the weather data. 
+  
+  my_url <- paste0("https://api.openweathermap.org/data/2.5/onecall?lat=", latitude, "&lon=", longitude,
+                  "&exclude=FALSE&appid=", apiKey)
+  
+  # the data are in JSON format, they have to be transformed before we can use them. 
+  my_raw_results <- httr::GET(my_url)
+  
+  my_content <- httr::content(my_raw_results, as = "text")
+  
+  my_content_from_json <- jsonlite::fromJSON(my_content)
+  
+  return(my_content_from_json)
+  
+}
+
+## Example for get_weather 
+get_weather("Amsterdam, Niederlande", Sys.getenv("MY_API"))
+
+
+# This function gets as input whether the user wants the current weather forecast, the hourly forecast, 
+# or the daily forecast, the location of interest and the apiKey. It returns a list with the desired wheather information.
+
+get_your_forecast <- function(cur_hour_day = current, location, apiKey) {
+  
+  full_weather <- get_weather(location, apiKey)
+  
+  if (cur_hour_day == "current") {
+    
+    weather <- full_weather$current 
+    
+  } else if (cur_hour_day == "hourly") {
+    
+    weather <- full_weather$hourly
+    
+  } else if (cur_hour_day == "daily") {
+    
+    weather <- full_weather$daily
+    
+  }
+  
+  return(weather)
+  
+}
+
+## Example for yourForecast()
+get_your_forecast("hourly", "Amsterdam, Niederlande", sys.getenv(MY_API))
+  
+
+# This function get a location (as a string) as input and returns and saves a map of this location using leaflet::leaflet().  
+
+get_map <- function(location) {
+  
+  latitude <- tidygeocoder::geo_osm(location)$lat[1]
+  longitude <- tidygeocoder::geo_osm(location)$long[1]

```

Comment:
see above; if the location cannot be geocoded you could show the user another png to indicate that